dev-tools/mage/copy.go

```diff
@@ -63,6 +70,15 @@ func (t *CopyTask) isExcluded(src string) bool {
 			return true
 		}
 	}
+
+	return false
+}
+
+func (t *CopyTask) isDestinationExcluded(dst string) bool {
```

Comment:
Is this function needed to exclude some paths? I remember seeing the same for the install flavors PR.
If this is the purpose shouldn't we try to do something that is useful in general?
See https://github.com/elastic/elastic-agent/issues/4588 for example.
Upcoming FIPS work will need some restriction on components included in packages as well.

cc @cmacknz

Reply:
Good point, this is excluding specific binaries which is exactly what the FIPS variant will require.

+1 to generalizing.

Reply:
this is checkFn not bound with flavors. you can pass whatever you like here. i do pass one from flavors to spare myself logic of parsing manifest and extracting paths out of it. but if you have checkFn provider for fips use case you can have that. you can have compundCheckFnProvider accepting more checks and returning skip if any of them fails. this can be enhanced. but please let's not do it here as it's not really related and i'm not sure how many `mage package` runs i have left

Reply:
guys have we reached a consensus here? maybe we should have a quick sync about it? 🙂

Reply:
The checkFn is not the ideal solution for the FIPS upcoming work but since this is under time constraints to get in 9.0, I guess we can merge it as-is and rework it later for FIPS

Reply:
cc @ebeahan